packages/flutter_tools/lib/src/android/android_device.dart

```diff
@@ -1019,6 +1020,7 @@ class AdbLogReader extends DeviceLogReader {
     }
     final Match timeMatch = AndroidDevice._timeRegExp.firstMatch(line);
     if (timeMatch == null || line.length == timeMatch.end) {
+      _acceptedLastLine = false;
```

Comment:
Without this change its really hard to stop the log reader with mocked output from just passing everything though